shell/browser/api/electron_api_app.cc

```diff
@@ -846,6 +846,7 @@ void App::SetAppPath(const base::FilePath& app_path) {
 #if !defined(OS_MACOSX)
 void App::SetAppLogsPath(gin_helper::ErrorThrower thrower,
                          base::Optional<base::FilePath> custom_path) {
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
```

Comment:
Hm, it's unclear to me which bit of this function does IO. Can we wrap this tighter by any chance? Put it closer to where the IO happens?

Reply:
base::PathService::Override checks if the directory exists (and thus hits the filesystem). I suppose i can wrap the call in a block and scope it there? but since it happens in both branches, that seemed excessive.

Reply:
It's best to scope this as narrowly as possible IMO, both because it reduces the likelihood we'll accidentally do some IO where we shouldn't be, and because it helps self-document (i.e. i wouldn't have asked this question if it was wrapped in a block with only one call to `PathService::Override`!)

Reply:
Good point! Fixed.